### DISCUSSION TITLE
How to download file/image  on message

### DISCUSSION THREAD

**Discussion by kingprapop** [Terrance/SkPy#253] [Q&A] [answered]
### Please anyone guide about ,how to download file/image on message that send from another person.
I try to catch message event :

`<URIObject uri="https://api.asm.skype.com/v1/objects/0-sa-d4-db5793e3a1cc89a6eb1bf99056235fb1" url_thumbnail="https://api.asm.skype.com/v1/objects/0-sa-d4-db5793e3a1cc89a6eb1bf99056235fb1/views/imgt1_anim" type="Picture.1" doc_id="0-sa-d4-db5793e3a1cc89a6eb1bf99056235fb1" width="1536" height="2048">when want to see picture click this link : <a href="https://login.skype.com/login/sso?go=xmmfallback?pic=0-sa-d4-db5793e3a1cc89a6eb1bf99056235fb1">https://login.skype.com/login/sso?go=xmmfallback?pic=0-sa-d4-db5793e3a1cc89a6eb1bf99056235fb1</a><OriginalName v="93D13B71-11B6-4965-A93F-FAF2FBBD28AD.png"></OriginalName><FileSize v="5339435"></FileSize><meta type="photo" originalName="93D13B71-11B6-4965-A93F-FAF2FBBD28AD.png"></meta></URIObject>`

My understand was IMAGE link .

` https://api.asm.skype.com/v1/objects/0-sa-d4-db5793e3a1cc89a6eb1bf99056235fb1/views/imgpsh_fullsize_anim`

but i can't authorize to save that file by python script.

### example code:

        if isinstance(event, SkypeNewMessageEvent) \
           and event.msg.userId != self.userId \
           and "url_thumbnail" in event.msg.content:
            cont_msg=event.msg.content          
            url=cont_msg[cont_msg.find("https"):cont_msg.find("url_thumbnail")-2]+"/views/imgpsh_fullsize_anim"
            url=url.replace("'","")
            if url is not None :
                response = requests.get(url)
                if response.status_code == 200:
                    image_data = response.content  # Get the binary image data
                    # Save the image data to a file
                    with open('temp.png', 'wb') as image_file:
                        image_file.write(image_data)

**Comment by Terrance** [comment, marked as answer]
As noted already in #252, you should be using [`SkypeFileMsg.fileContent`](https://skpy.t.allofti.me/reference/api.html#skpy.msg.SkypeFileMsg.fileContent):

```python
with open('temp.png', 'wb') as image_file:
    image_file.write(event.msg.fileContent)
```

If that's not retrieving the content you expect, it would be useful to know what you _are_ expecting instead.

If a file offers multiple URLs and you need to fetch a different one, see the definition of `fileContent` for how it authenticates:

https://github.com/Terrance/SkPy/blob/00797b16a656ae13355c356d514feb0bdeb40579/skpy/msg.py#L602-L603

You should be able to pass in an alternate URL and have it authenticate in the same way.  (See also: [`Auth.Authorize`](https://skpy.t.allofti.me/reference/internal.html#skpy.conn.SkypeConnection.Auth))

**Reply by kingprapop** [reply]
I will try that on Monday.
thank you for good explain.
